src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -177,8 +178,56 @@ class RowGenotype(r: Row) extends Genotype {
 
 object Genotype {
   def buildGenotypeExtractor(t: Type): (Any) => Genotype = {
-    assert (t == TGenotype)
-    (a: Any) => a.asInstanceOf[Genotype]
+    if (t == TGenotype)
+      (a: Any) => a.asInstanceOf[Genotype]
+    else {
+      def queryOpt[T >: Null](field: String)(implicit hr: HailRep[T]): Option[(Any) => T] = {
+        try {
+          val (ft, fq0) = t.queryTyped(field)
+          if (ft == hr.typ)
+            Some((a: Any) => fq0(a).asInstanceOf[T])
+          else
+            None
+        } catch {
+          case _: AnnotationPathException =>
+            None
+        }
+      }
+
+      val gtq = queryOpt[java.lang.Integer]("GT")(callHr)
+      val adq = queryOpt[IndexedSeq[Int]]("AD")
+      val dpq = queryOpt[java.lang.Integer]("DP")
+      val gqq = queryOpt[java.lang.Integer]("GQ")
+      val plq = queryOpt[IndexedSeq[Int]]("PL")
+      val gpq = queryOpt[IndexedSeq[Double]]("GQ")
+
+      if (plq.isDefined || gpq.isEmpty)
+        (a: Any) =>
+          new RowGenotype(Row(
+            gtq.map(_ (a)).orNull,
+            adq.map(_ (a)).orNull,
+            dpq.map(_ (a)).orNull,
+            gqq.map(_ (a)).orNull,
+            plq.map(_ (a)).orNull,
+            false,
+            false))
+      else
+        (a: Any) => {
+          val gp = gpq.map(_ (a)).orNull
+
+          new RowGenotype(Row(
+            gtq.map(_ (a)).orNull,
+            adq.map(_ (a)).orNull,
+            dpq.map(_ (a)).orNull,
+            gqq.map(_ (a)).orNull,
+            if (gp != null)
+              Genotype.weightsToLinear(gp.toArray)
```

Comment:
The fifth row element for RowGenotype is PX, which is expected to be an `IndxedSeq[Int]`:

```
  def _unboxedPX: Array[Int] =
    if (r.isNullAt(4))
      null
    else
      r.getSeq(4).toArray
```